### DISCUSSION TITLE
grons for Pekko (open source akka)

### DISCUSSION THREAD

**Discussion by normana400** [erikvanoosten/metrics-scala#481] [Ideas]
now that the Pekko (aka open source) fork of akka is live (latest version is currently 1.0.1), is there a planned implementation of support for pekko.
essentially it should work like the metrics4 akka lib (but with different artifacts / package names)

https://pekko.apache.org/docs/pekko/current/project/migration-guides.html

**Comment by erikvanoosten** [comment]
Hi @normana400 . If there is interest, a Pekko version can be created for sure!

I have not used metrics-scala in any project for over 5 years, so it is not high on my priority list. I would definitely support a pull request though.

**Reply by normana400** [reply]
https://github.com/erikvanoosten/metrics-scala/pull/483

**Reply by erikvanoosten** [reply]
Looks good! I'll take a more thorough look tomorrow.

**Comment by erikvanoosten** [comment]
This is included in version 4.3.0 that was just released. 🥳

We need to figure out the compatibility rules of Pekko to update some names. But this is not urgent.

**Reply by normana400** [reply]
class names are the same. 
every package name starting with "com.typesafe.akka..." or "akka" now is "org.apache.pekko..."
that idea also pertains to artifact names except for kafka streams / alpakka which is named: "pekko-connectors-kafka"

typesafe configs that were akka{} are now pekko{}

I think that is the gist of the dfferences in naming

**Reply by erikvanoosten** [reply]
I meant compatibility rules between different pekko versions 😄

**Reply by erikvanoosten** [reply]
... as those determine the versioning scheme for metrics-pekko.

**Comment by sylvoutt** [comment]
sorry for my answer, i need your version with pekko but you didn't publish it on maven like the other. It's a choice ?

**Comment by erikvanoosten** [comment]
Ehh, ☺️
It seems you are correct! I'll take a look when I am back from my holidays.

**Comment by erikvanoosten** [comment]
I am back from my holiday. I did some research but I do not understand what went wrong at all.

Unfortunately, my sonatype account got disabled. As soon as that is fixed I will do a new release, this time hopefully including the pekko module. Meanwhile, I will create a new issue.

**Comment by erikvanoosten** [comment]
Now that the situation is resolved I can write a bit more. My Sonatype account got disabled because of MavenGate. It is not clear why `nl.grons` came on the 'vulnerability' list though. My current assumption is that it was a mistake. I wrote a bit about the situation on https://day-to-day-stuff.blogspot.com/2024/08/mavengate-gets-it-all-wrong-and-hurts.html

This does not explain why the previous artifacts are missing in maven central. Unfortunately, I do not know of a way to investigate this further so I am leaving it at that.

cc: @sylvoutt

**Comment by sylvoutt** [comment]
@erikvanoosten  Thanks a lot
